Program.cs

```diff
@@ -19,13 +19,53 @@ public static List<string> ReadWordsFromFile(string filePath)
         return words;
     }
 
-    public static void Main()
+    public static List<string> RemoveDuplicateWords(List<string> words)
     {
-        List<string> wordsList = ReadWordsFromFile("C:\\Users\\Zoran\\Desktop\\Dario_Internship\\#3_Words\\text.txt");
+        HashSet<string> uniqueWords = new HashSet<string>(words);
+        return new List<string>(uniqueWords);
+    }
 
-        foreach (string word in wordsList)
+    public static Dictionary<string, int> CountWordFrequency(List<string> words)
+    {
+        Dictionary<string, int> wordFrequency = new Dictionary<string, int>(); 
+        foreach (string word in words)
         {
-            Console.WriteLine(word);
+            if (wordFrequency.ContainsKey(word)) wordFrequency[word]++;
+            else wordFrequency[word] = 1;
         }
+        return wordFrequency;
+    }
+
+    public static void PrintSummary(List<string> wordsList, Dictionary<string, int> wordFrequency)
+    {
+        int totalWords = wordsList.Count;
+        int uniqueWords = wordFrequency.Count;
+
+        Console.WriteLine($"Total number of words in the document: {totalWords}");
+        Console.WriteLine($"Number of unique words: {uniqueWords}");
+
+        int maxFrequency = wordFrequency.Values.Max();
+        List<string> mostFrequentWords = wordFrequency.Where(pair => pair.Value == maxFrequency)
+                                                     .Select(pair => pair.Key)
+                                                     .ToList();
```

Comment:
It would be great if you could answer the following questions, as of now the implementation is working completely fine!
- This covers an edge case with the word counts. Could you explain what that case is?
- Could you provide an alternative implementation without the LINQ methods (`.Where`, `.Select`, `.ToList`)?
- Could you compare the `.Where` to a existing JS function?
- Could you compare the `.Select` to a existing JS function as well?
- Could you provide the _first most occuring word_ using LINQ?

Reply:
- Edge case is there is more than one pair with maximum value
- I could've used foreach loop to loop through wordFrequency List. In there if pair.value equals maxFrequency I would add it to the mostFrequentWords list. 
-   I could compare` .Where` to `filter()` JS function as they both filter certain members based on condition. Syntax is also the same
- The `.Select` is like `.map()` in JavaScript as they both transform each element of the List by applying certain operation to them. 
- Most frequent word(s) with frequency 6: vitae